docs/ref/models/fields.txt

```diff
@@ -1630,95 +1630,30 @@ Django also defines a set of fields that represent relations.
 
-.. _recursive-relationships:
-
-To create a recursive relationship -- an object that has a many-to-one
-relationship with itself -- use ``models.ForeignKey('self',
-on_delete=models.CASCADE)``.
-
-.. _lazy-relationships:
-
-If you need to create a relationship on a model that has not yet been defined,
-you can use the name of the model, rather than the model object itself::
-
-    from django.db import models
-
-
-    class Car(models.Model):
-        manufacturer = models.ForeignKey(
-            "Manufacturer",
-            on_delete=models.CASCADE,
-        )
-        # ...
-
-
-    class Manufacturer(models.Model):
-        # ...
-        pass
-
-Relationships defined this way on :ref:`abstract models
-<abstract-base-classes>` are resolved when the model is subclassed as a
-concrete model and are not relative to the abstract model's ``app_label``:
-
-.. code-block:: python
-    :caption: ``products/models.py``
-
-    from django.db import models
-
-
-    class AbstractCar(models.Model):
-        manufacturer = models.ForeignKey("Manufacturer", on_delete=models.CASCADE)
-
-        class Meta:
-            abstract = True
-
-.. code-block:: python
-    :caption: ``production/models.py``
-
-    from django.db import models
-    from products.models import AbstractCar
-
-
-    class Manufacturer(models.Model):
-        pass
-
-
-    class Car(AbstractCar):
-        pass
-
-
-    # Car.manufacturer will point to `production.Manufacturer` here.
-
-To refer to models defined in another application, you can explicitly specify
-a model with the full application label. For example, if the ``Manufacturer``
-model above is defined in another application called ``production``, you'd
-need to use::
-
-    class Car(models.Model):
-        manufacturer = models.ForeignKey(
-            "production.Manufacturer",
-            on_delete=models.CASCADE,
-        )
-
-This sort of reference, called a lazy relationship, can be useful when
-resolving circular import dependencies between two applications.
+The first positional argument can be either a concrete model class or a
+:ref:`lazy reference <lazy-relationships>` to a model class.
+:ref:`Recursive relationships <recursive-relationships>`, where a model has a
+relationship with itself, are also supported.
 
-A database index is automatically created on the ``ForeignKey``. You can
-disable this by setting :attr:`~Field.db_index` to ``False``.  You may want to
-avoid the overhead of an index if you are creating a foreign key for
-consistency rather than joins, or if you will be creating an alternative index
-like a partial or multiple column index.
+See :attr:`ForeignKey.on_delete` for details on the second positional
+argument.
 
 Database Representation
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 Behind the scenes, Django appends ``"_id"`` to the field name to create its
 database column name. In the above example, the database table for the ``Car``
```

Comment:
With the extracted code, this phrase is now confusing, because it references a `Car` class that hasn't been created nor discussed so far. Would you have ideas in how to approach this?

One initial option I could think of is leaving an example in place for using an FK with a `Car` model, but without using any sort of recursion nor lazy reference.

Reply:
Good catch! I missed that